### PR TITLE
Update clusterclasses to use TKR_DATA variable values to populate image data

### LINKED ISSUE #1993: Store TKR Resolution results in Cluster variables
Add a ClusterClass variable named `TKR_DATA`. The value of this variable will contain a mapping from Kubernetes version strings (resolved values of `spec.topology.version`) to a map containing the data from the resolved TKR and OSImage:

```yaml
v1.22.3+vmware.1:
  kubernetesSpec:
    version: v1.22.3+vmware.1
    imageRepository: projects-stg.registry.vmware.com/tkg
    etcd:
      imageTag:  v3.5.0_vmware.7
    pause:
      imageTag: "3.5"
    coredns:
      imageTag: v1.8.4_vmware.7
    kube-vip:
      imageTag: v0.4.2_vmware.7
  osImageRef:
    id: ami-0f2e5eec7ae0a1986
    region: ap-northeast-1
  labels:
    run.tanzu.vmware.com/tkr: v1.22.3---vmware.1-tkg.1
    run.tanzu.vmware.com/os-image: v1.22.3---vmware.1-tkg.1-ami-0f2e5eec7ae0a1986
    os-type: linux
    os-name: ubuntu
    os-version: "20.04"
    os-arch: amd64
    image-type: ami
    ami-id: ami-0f2e5eec7ae0a1986
    ami-region: ap-northeast-1
v1.22.4+vmware.1:
  kubernetesSpec:
    version: v1.22.4+vmware.1
    imageRepository: projects-stg.registry.vmware.com/tkg
    etcd:
      imageTag:  v3.5.0_vmware.7
    pause:
      imageTag: "3.5"
    coredns:
      imageTag: v1.8.4_vmware.7
    kube-vip:
      imageTag: v0.4.2_vmware.7
  osImageRef:
    id: ami-0a3d6eac8ae0a2387
    region: ap-northeast-1
  labels:
    run.tanzu.vmware.com/tkr: v1.22.4---vmware.1-tkg.1
    run.tanzu.vmware.com/os-image: v1.22.4---vmware.1-tkg.1-ami-0a3d6eac8ae0a2387
    os-type: linux
    os-name: ubuntu
    os-version: "20.04"
    os-arch: amd64
    image-type: ami
    ami-id: ami-0a3d6eac8ae0a2387
    ami-region: ap-northeast-1
```

### LINKED ISSUE #2171: Update Clusterclasses to consume new resolved data from TKR_DATA
(This is used to request new product features)

**Describe the feature request**
Clusterclasses should uses values set in TKR_DATA, which is set by the TKR resolver, to populate image fields in KubeadmControlPlaneTemplate

**Describe alternatives you've considered**

**Affected product area (please put an X in all that apply)**

- [ ] APIs
- [ ] Addons
- [x] CLI
- [ ] Docs
- [ ] IAM
- [x] Installation
- [ ] Plugin
- [ ] Security
- [ ] Test and Release
- [ ] User Experience

**Additional context**
See #1993 for structure of TKR_DATA values

### PR DESCRIPTION
**Pull request by tenczar**
This change updates the kubeletExtraArgs to set node-labels based on tkrData. Also updates the KubeadmControlPlaneTemplate to populate etcd and coredns image data from the tkr variable.

### What this PR does / why we need it
Updates clusterclass definitions to consume tkrData to set image repository and tag data.

### Which issue(s) this PR fixes

Fixes #2171  

### Describe testing done for PR

Tested all template changes with a template evaluator tool. Confirmed that the generated templates have the desired values set.

### Release note

```release-note

```

### PR Checklist

- [ ] Squash the commits into one or a small number of logical commits
      
- [ ] Use good commit [messages](https://github.com/vmware-tanzu/tanzu-framework/blob/main/CONTRIBUTING.md)
- [ ] Ensure PR contains terms all contributors can understand and links all contributors can access

### CODE REVIEW

File: pkg/v1/providers/infrastructure-aws/v1.2.0/cconly/base.yaml [hunk 1]
```diff
@@ -987,16 +1059,16 @@ spec:
               cloud-provider: aws
               tls-cipher-suites: TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256,TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256,TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305,TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384,TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305,TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384
           dns:
-            imageRepository: projects-stg.registry.vmware.com/tkg #! TODO '${_TKG_COREDNS_IMAGE_REPOSITORY}' May be a variable for package level
-            imageTag: v1.8.4_vmware.5 #! TODO '${_TKG_COREDNS_IMAGE_TAG}' May be a variable for package level
+            imageRepository: projects-stg.registry.vmware.com/tkg
```
Comment (vuil) [inline pkg/v1/providers/infrastructure-aws/v1.2.0/cconly/base.yaml:1062]: nit: populating with real looking but likely incorrect values only to be patched is misleading. Can we not set it at all? If we want to suggest example values, perhaps do something like:
```
imageTag: #! e.g v3.5.0_vmware.5
```

File: pkg/v1/providers/infrastructure-aws/v1.2.0/cconly/base.yaml [hunk 2]
```diff
@@ -307,10 +307,11 @@ spec:
       openAPIV3Schema:
         type: boolean
         default: false
-  - name: TKR_KUBERNETES_SPEC
+  - name: tkrData
     schema:
       openAPIV3Schema:
         type: object
+        properties: {}
```
Comment (vuil) [inline pkg/v1/providers/infrastructure-aws/v1.2.0/cconly/base.yaml:314]: @imikushin what's the recommendation in terms of definition of the TKR_DATA, to keep as generic object? for now?

File: pkg/v1/providers/infrastructure-aws/v1.2.0/cconly/base.yaml [hunk 3]
```diff
@@ -525,6 +554,49 @@ spec:
         path: "/spec/template/spec/rootVolume/size"
         valueFrom:
           variable: AWS_NODE_OS_DISK_SIZE_GIB
+
+  - name: AWS_NODE_LABELS
```
Comment (vuil) [inline pkg/v1/providers/infrastructure-aws/v1.2.0/cconly/base.yaml:558]: nit: I don't see UPPER_SNAKE casing used in naming patches else where in the definition

**Comment by vuil** [comment]
Can you describe a little more how the testing was done? What test variable and value did you use to test the patches?

File: pkg/v1/providers/infrastructure-aws/v1.2.0/cconly/base.yaml (hunk 2 above)
Comment (imikushin) [reply pkg/v1/providers/infrastructure-aws/v1.2.0/cconly/base.yaml:314]: For now, yes. We could also add this to openAPIV3Schema (bc it's basically a map with arbitrary keys):
```yaml
x-kubernetes-preserve-unknown-fields: true
```

The schema is basically defined here: https://github.com/vmware-tanzu/tanzu-framework/blob/main/pkg/v2/tkr/webhook/cluster/tkr-resolver/cluster/tkrdata.go

If folks have time to translate this to openAPIV3Schema, that would be great.

**Comment by anujc25** [comment]
Lint pipeline is failing with
```
/tanzu-framework/pkg/v1/providers/infrastructure-azure/v1.0.1/cconly/base.yaml
  594:19    error    trailing spaces  (trailing-spaces)
```

File: pkg/v1/providers/infrastructure-azure/v1.0.1/yttcc/overlay.yaml [hunk 4]
```diff
@@ -110,7 +110,27 @@ spec:
...
+          labels:
+            os-name: ubuntu
+            os-type: linux
+            os-arch: amd64
+          osImageRef:
+            sku: spec.image.marketplace.sku
+            publisher: spec.image.marketplace.publisher
+            offer: spec.image.marketplace.offer
+            version: spec.image.marketplace.version
+            thirdPartyImage: spec.image.marketplace.thirdPartyImage
```
Comment (anujc25) [inline pkg/v1/providers/infrastructure-azure/v1.0.1/yttcc/overlay.yaml:133]: As we are using this as a dummy data, can we use more accurate dummy values to reduce the confusion for reviewer?

**Review by anujc25 — APPROVED** [review]
One minor comment but otherwise looks good.

**Comment by vuil** [comment]
Please review https://storage.googleapis.com/tkg-clustergen/2161/20220503000725/clustergen_cc.diff.txt as well
(the pr bot automation cannot be updated until pblcm is merged back to main)

File: pkg/v1/providers/infrastructure-aws/v1.2.0/yttcc/overlay.yaml [hunk 5]
```diff
@@ -100,3 +100,22 @@ spec:
       value: #@ vars[configVariable]
     #@ end
     #@ end
+    #@overlay/append
+    - name: TKR_DATA
```
Comment (vuil) [inline pkg/v1/providers/infrastructure-aws/v1.2.0/yttcc/overlay.yaml:104]: please add a comment that this should be removed (only conditionally included in test scenarios)

File: pkg/v1/providers/infrastructure-aws/v1.2.0/cconly/base.yaml (hunk 2 above)
Comment (vuil) [reply pkg/v1/providers/infrastructure-aws/v1.2.0/cconly/base.yaml:314]: @tenczar assuming we are not doing this in this PR, can you file an issue to track possibly defining a more specific schema for this variable.

**Review by anujc25 — APPROVED** [review]
Looks good to me.

**Pull request merged**